types/three/three-core.d.ts

```diff
@@ -3391,6 +3391,7 @@ export class Color {
     equals(color: Color): boolean;
     fromArray(rgb: number[], offset?: number): this;
     toArray(array?: number[], offset?: number): number[];
+    toArray(xyz: ArrayLike<number>, offset?: number): ArrayLike<number>;
```

Comment:
Should `xyz` be optional? Is it possible to fold these into single overloads using union types?

Reply:
I see - perhaps

```
<T extends ArrayLike<number> | number[]>(toArray: T, offset?: number): T;
```

Reply:
Hi Daniel,

maybe we can have a look at following example:
        let arrayBuffer = new Float32Array(3);
        let vec = new Vector3();

        let xyz = vec.toArray(arrayBuffer);
        xyz.push(42); //Error as expected b/c xyz is pointing to arrayBuffer

        let arr = vec.toArray();
        arr.push(42); //No error b/c THREE creates a number[] internaly

With my change I get an error as expected in line 2 but no error in line 5.
Applying your changes I get errors in line 2 and line 5 thus limiting the users flexibility.

To remove the error in line 5 one could explicetely specify T:
`let arr = vec.toArray<number[]>();`
`arr.push(42); //No error b/c THREE creates a number[] internaly`

But I doubt that this is what you had in mind.

So as mentioned before, I see no way arround defining the method twice.